core/mount/mount_linux.go

```diff
@@ -250,38 +248,75 @@ func getUnprivilegedMountFlags(path string) (int, error) {
 }
 
 func doPrepareIDMappedOverlay(tempRemountsLocation string, lowerDirs []string, usernsFd int) ([]string, func(), error) {
-	tmpLowerDirs := make([]string, 0, len(lowerDirs))
+	commonDir, err := getCommonDirectory(lowerDirs)
+	if err != nil {
```

Comment:
I'd return an error also if commonDir is not inside the root path in the config toml. Or is it being overly cautious?

Reply:
Fair, if we can't access that here, that is a great reason :D